libs/langchain/tests/unit_tests/document_transformers/test_beautiful_soup_transformer.py

```diff
@@ -15,36 +15,151 @@ def test_transform_empty_html() -> None:
 
 
 @pytest.mark.requires("bs4")
-def test_extract_paragraph() -> None:
+def test_extract_paragraphs() -> None:
     bs_transformer = BeautifulSoupTransformer()
-    paragraphs_html = "<html><p>First paragraph.</p><p>Second paragraph.</p><html>"
+    paragraphs_html = (
+        "<html><h1>Header</h1><p>First paragraph.</p><p>Second paragraph.</p><html>"
+    )
+    documents = [Document(page_content=paragraphs_html)]
+    docs_transformed = bs_transformer.transform_documents(documents)
+    assert docs_transformed[0].page_content == "First paragraph. Second paragraph."
+
+
+@pytest.mark.requires("bs4")
+def test_strip_whitespace() -> None:
+    bs_transformer = BeautifulSoupTransformer()
+    paragraphs_html = (
+        "<html><h1>Header</h1><p><span>First</span>   paragraph.</p>"
+        "<p>Second paragraph. </p><html>"
+    )
     documents = [Document(page_content=paragraphs_html)]
     docs_transformed = bs_transformer.transform_documents(documents)
     assert docs_transformed[0].page_content == "First paragraph. Second paragraph."
 
 
+@pytest.mark.requires("bs4")
+def test_extract_html() -> None:
+    bs_transformer = BeautifulSoupTransformer()
+    paragraphs_html = (
+        "<html>Begin of html tag"
+        "<h1>Header</h1>"
+        "<p>First paragraph.</p>"
+        "Middle of html tag"
+        "<p>Second paragraph.</p>"
+        "End of html tag"
+        "</html>"
+    )
+    documents = [Document(page_content=paragraphs_html)]
+    docs_transformed = bs_transformer.transform_documents(
+        documents, tags_to_extract=["html", "p"]
+    )
+    assert docs_transformed[0].page_content == (
+        "Begin of html tag "
+        "Header First paragraph. "
+        "Middle of html tag "
+        "Second paragraph. "
+        "End of html tag"
+    )
+
+
 @pytest.mark.requires("bs4")
 def test_remove_style() -> None:
     bs_transformer = BeautifulSoupTransformer()
     with_style_html = (
         "<html><style>my_funky_style</style><p>First paragraph.</p></html>"
     )
     documents = [Document(page_content=with_style_html)]
-    docs_transformed = bs_transformer.transform_documents(documents)
+    docs_transformed = bs_transformer.transform_documents(
+        documents, tags_to_extract=["html"]
```

Comment:
Why does this test change?

Reply:
Thanks for the fast review and the collaboration on the fix :-) 

This change fixes a bug in the test. In the previous version, the test was actually not testing "unwanted_tags" functionality.

The `my_funky_style` was not in the output, but _not_ because it was filtered out as unwanted_tag, but simply since it was not included one of `tags_to_extract` (default `p` etc. but _not_ `html`).

Proof is that this variation of the test below is green (while it should be red, since `style` is not part of the `unwanted_tags` in this variation):

_Green_ test (which it should not be in the logic of the test):

```
   with_style_html = (
        "<html><style>my_funky_style</style><p>First paragraph.</p></html>"
    )
    documents = [Document(page_content=with_style_html)]
    docs_transformed = bs_transformer.transform_documents(
        documents, unwanted_tags=["script"],  # removing "style" from the default list
    )
    assert docs_transformed[0].page_content == "First paragraph."
```

This test, with `html` added to `tags_to_extract`  is correctly RED:

```
    with_style_html = (
        "<html><style>my_funky_style</style><p>First paragraph.</p></html>"
    )
    documents = [Document(page_content=with_style_html)]
    docs_transformed = bs_transformer.transform_documents(
        documents, unwanted_tags=["script"], tags_to_extract=["html"]
    )
    assert docs_transformed[0].page_content == "First paragraph."
```

Now (correctly) leads to the expected error:

```
E       AssertionError: assert 'my_funky_style First paragraph.' == 'First paragraph.'
E         - First paragraph.
E         + my_funky_style First paragraph.
```

And becomes green again, by actually adding `style` back to the unwanted_tags;
which is the current version in the PR.